doc/api.rst

```diff
@@ -14,27 +14,35 @@ generator.py
     .. cmdoption:: -s <script>
     .. cmdoption:: -d <directory>
     .. cmdoption:: -l
-
-    Command line version of the generator. -h prints out the command
-    line options. If -oalg or -opsy are not provided then the
-    generated code is printed to stdout, otherwise they are output to
-    the specified file name. -api specifies the particular api to
-    use. -s allows a script to be called which can modify (typically
-    optimise) the PSy layer. -d specifies a directory to recursively
-    search to find the associated kernel files. -l limits the maximum
-    line length of the fortran output to 132 characters. -l uses a
-    relatively simple algorithm which in pathological cases may
-    produce incorrect output, so it is recommended to only use this
-    option if necessary. generator.py Uses the
-    :func:`generator.generate` function to generate the code. Please
-    see the run documentation for more details.
-
-    For example::
-
-    > python generator.py algspec.f90
-    > python generator.py -oalg alg.f90 -opsy psy.f90 -api dynamo0.3 algspec.f90
-    > python generator.py -d ../kernel -s opt.py algspec.f90
-    > python generator.py -s ../scripts/opt.py -l algspec.f90
+    .. cmdoption:: -dm/-nodm
+    .. cmdoption:: --profile/--force-profile <invokes,kernels>
+
+Command line version of the generator. -h prints out the command
+line options. If -oalg or -opsy are not provided then the
+generated code is printed to stdout, otherwise they are output to
+the specified file name. -api specifies the particular api to
+use. -s allows a script to be called which can modify (typically
+optimise) the PSy layer. -d specifies a directory to recursively
+search to find the associated kernel files. -l limits the maximum
+line length of the fortran output to 132 characters. -l uses a
+relatively simple algorithm which in pathological cases may
+produce incorrect output, so it is recommended to only use this
+option if necessary. -dm/-nodm enables/disables the generation of
+distributed-memory code (if the selected API supports this). --profile
+enables the automatic insertion of profiling calls at either the
+invoke or kernel level. `psyclone` uses the
+:func:`generator.generate` function to generate the code. Please
+see the :ref:`psyclone_script` documentation for more details.
```

Comment:
Looking at PDF output, I think that perhaps it would be visually easier to follow this paragraph if it was formulated like a list.

Reply:
In fact, rather than put effort into this, I've removed this sub-section entirely on the grounds that the script isn't part of the API *and* it already has its own section :-)